lib/Mailer/SparkPost.php

```diff
@@ -0,0 +1,131 @@
+<?php
+
+namespace Stampie\Mailer;
+
+use Stampie\Adapter\ResponseInterface;
+use Stampie\Attachment;
+use Stampie\Exception\ApiException;
+use Stampie\Exception\HttpException;
+use Stampie\Mailer;
+use Stampie\Message\AttachmentsAwareInterface;
+use Stampie\Message\MetadataAwareInterface;
+use Stampie\Message\TaggableInterface;
+use Stampie\MessageInterface;
+use Stampie\Util\IdentityUtils;
+
+/**
+ * Mailer to be used with SparkPost's API.
+ *
+ * @author Shaun Simmons <shaun@envysphere.com>
+ */
+class SparkPost extends Mailer
+{
+    protected function getEndpoint()
+    {
+        return 'https://api.sparkpost.com/api/v1/transmissions';
+    }
+
+    protected function getHeaders()
+    {
+        return [
+            'Authorization' => $this->getServerToken(),
+        ];
+    }
+
+    protected function format(MessageInterface $message)
+    {
+        $from = $this->normalizeIdentity($message->getFrom());
+
+        $parameters = [
+            'options' => [
+                'transactional' => true,
```

Comment:
Is there a clean way of making this configurable?

Reply:
I suggest leaving it out for now. And then, we could make it a configuration setting of the mailer itself.

If you want to send both transactional and marketing emails, you would then have to instantiate 2 different mailers

Reply:
I've added an argument to the mailer constructor to control this parameter.